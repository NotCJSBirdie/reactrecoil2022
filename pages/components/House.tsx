import React from "react";
import { useState } from "react";

const House = () => {
  const [house, setHouse] = useState(false);

  const changeHouse = () => {
    setHouse(!house);
  };

  return (
    <div>
      <div className="flex flex-row items-center border-4 border-orange-400 w-full">
        {house ? (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="icon icon-tabler icon-tabler-home"
            width="64"
            height="64"
            viewBox="0 0 24 24"
            strokeWidth="1.5"
            stroke="#6f32be"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path stroke="none" d="M0 0h24v24H0z" fill="none" />
            <polyline points="5 12 3 12 12 3 21 12 19 12" />
            <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
            <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
          </svg>
        ) : (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="icon icon-tabler icon-tabler-home"
            width="64"
            height="64"
            viewBox="0 0 24 24"
            strokeWidth="1.5"
            stroke="#00b341"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path stroke="none" d="M0 0h24v24H0z" fill="none" />
            <polyline points="5 12 3 12 12 3 21 12 19 12" />
            <path d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7" />
            <path d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6" />
          </svg>
        )}
        <button
          onClick={changeHouse}
          className="bg-red-400 text-white p-8 text-center my-8"
        ></button>
      </div>
    </div>
  );
};

export default House;
