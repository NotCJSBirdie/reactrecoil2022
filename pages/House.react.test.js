import React from "react";
import renderer from "react-test-renderer";
import House from "./components/House";

test("House changes when button is clicked", () => {
  const component = renderer.create(<House />);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
