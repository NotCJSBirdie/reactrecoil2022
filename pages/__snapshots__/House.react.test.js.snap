// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`House changes when button is clicked 1`] = `
<div>
  <div
    className="flex flex-row items-center border-4 border-orange-400 w-full"
  >
    <svg
      className="icon icon-tabler icon-tabler-home"
      fill="none"
      height="64"
      stroke="#00b341"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <polyline
        points="5 12 3 12 12 3 21 12 19 12"
      />
      <path
        d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
      />
      <path
        d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
      />
    </svg>
    <button
      className="bg-red-400 text-white p-8 text-center my-8"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`House changes when button is clicked 2`] = `
<div>
  <div
    className="flex flex-row items-center border-4 border-orange-400 w-full"
  >
    <svg
      className="icon icon-tabler icon-tabler-home"
      fill="none"
      height="64"
      stroke="#00b341"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <polyline
        points="5 12 3 12 12 3 21 12 19 12"
      />
      <path
        d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
      />
      <path
        d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
      />
    </svg>
    <button
      className="bg-red-400 text-white p-8 text-center my-8"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`Link changes the class when hovered 1`] = `
<div>
  <div
    className="flex flex-row items-center border-4 border-orange-400 w-full"
  >
    <svg
      className="icon icon-tabler icon-tabler-home"
      fill="none"
      height="64"
      stroke="#00b341"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1.5"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <polyline
        points="5 12 3 12 12 3 21 12 19 12"
      />
      <path
        d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
      />
      <path
        d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
      />
    </svg>
    <button
      className="bg-red-400 text-white p-8 text-center my-8"
      onClick={[Function]}
    />
  </div>
</div>
`;
